File: lesson_22_java/src/main/java/com/lesson22java/entities/Product.java
package com.lesson22java.entities;


import jakarta.persistence.*;
import lombok.Data;

//CREATE TABLE product (
//        id bigserial NOT NULL,
//        title text NULL,
//        price int4 NULL,
//        CONSTRAINT product_pk PRIMARY KEY (id)
//);
@Entity
@Table(name = "product")
@Data
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;
    @Column(name = "title")
    private String title;
    @Column(name = "price")
    private Integer price;
}

File: lesson_20/src/main/kotlin/lesson20kt/entities/Product.kt
package lesson20kt.entities

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "product")
data class Product(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var id: Long? = null,

    @Column(name = "title")
    var title: String? = null,

    @Column(name = "price")
    var price: Int? = null,
)